src/app/component/layout/components/search-popup/search-popup.component.spec.ts

```diff
@@ -0,0 +1,120 @@
+import { MatDialogModule } from '@angular/material/dialog';
+import { SearchService } from '@global-service/search/search.service';
+import { SearchModel } from '@global-models/search/search.model';
+import { NewsSearchModel } from '@global-models/search/newsSearch.model';
+import { Observable } from 'rxjs';
+import { NgxPageScrollModule } from 'ngx-page-scroll';
+import { ReactiveFormsModule } from '@angular/forms';
+
+describe('SearchPopupComponent', () => {
+  let component: SearchPopupComponent;
+  let fixture: ComponentFixture<SearchPopupComponent>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        SearchPopupComponent,
+        SearchItemComponent,
+        SearchNotFoundComponent,
+      ],
+      imports: [
+        RouterTestingModule,
+        TranslateModule.forRoot(),
+        ReactiveFormsModule,
+        HttpClientTestingModule,
+        MatDialogModule,
+        NgxPageScrollModule,
+      ],
+      providers: [
+        SearchService
+      ]
+    })
+      .compileComponents().then(r => r);
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(SearchPopupComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create SearchPopupComponent', () => {
+    expect(component).toBeTruthy();
+  });
+
+  describe('General methods', () => {
+
+    it(`ngOnInit should init setupInitialValue method`, () => {
+        const spy = spyOn(component as any, 'setupInitialValue');
+        component.ngOnInit();
+        expect(spy).toHaveBeenCalledTimes(1);
+      });
+
+    it('should call openErrorPopup', () => {
+      spyOn(component.dialog, 'open');
+      component.openErrorPopup();
+      expect(component.dialog).toBeDefined();
+    });
+
+  });
+
+  describe('Testing services:', () => {
+    let searchService: SearchService;
+    let mockDataSearchModel: SearchModel;
+    let mockNewsSearchModel: NewsSearchModel;
+
+    beforeEach(() => {
+      searchService = fixture.debugElement.injector.get(SearchService);
+
+      mockDataSearchModel = {
+        countOfResults: 4,
+        ecoNews: [mockNewsSearchModel],
+        tipsAndTricks: [mockNewsSearchModel]
+   };
+
+      mockNewsSearchModel = {
+     id: 10,
+      title: 'taras',
+      author: {
+        id: 20,
+        name: 'Ivan',
+      },
+      creationDate: 'data-time',
+      tags: ['news'],
+   };
+    });
+
+    it('should handle search value changes', () => {
+      const getSearchSpy = spyOn(searchService, 'getSearch').and.returnValue(Observable.of(mockDataSearchModel));
+      const resetDataSpy = spyOn(component as any, 'resetData');
+      component.ngOnInit();
+
+      component.searchInput.setValue('test', { emitEvent: true });
+      expect(getSearchSpy).toHaveBeenCalledWith('test');
+
+      component.searchInput.setValue('', { emitEvent: true });
+      expect(resetDataSpy).toHaveBeenCalled();
```

Comment:
Best practice when for one "it" only one "expect"

Reply:
fixed